### ISSUE TITLE
mock_all_from_stat calls the mock sub twice

### ISSUE THREAD

**Issue by toddr** [CpanelInc/Overload-FileCheck#4]
```
$>perl -MOverload::FileCheck=:all -E'sub mock {say shift; return [0..12]}; mock_all_from_stat(\&mock); -s "/abc"'     
lstat
lstat
```

**Comment by toddr** [comment]
This doesn't break anything but it seems strange. Is there a reason?

**Comment by atoomic** [comment]
the reason is that we need to know if the the file is mocked or doing a fallback
I've a pending optimization to avoid to expose the second call, but yes until we have a better framework they are required

**Comment by atoomic** [comment]
closed by e3c04893aa18878fa63fe5ade7aa72d7299edd7c
